Sources/TuistSupport/Utils/Environment.swift

```diff
@@ -173,4 +197,26 @@ public class Environment: Environmenting {
     public var settingsPath: AbsolutePath {
         directory.appending(component: "settings.json")
     }
+    
+    private func loadEnvironmentVariablesFromFile() throws {
+        guard FileManager.default.fileExists(atPath: ".env") else {
+            throw EnvFileFailure.fileMissing(path: ".env")
+        }
+        guard let contents = try? String(contentsOf: URL(fileURLWithPath: ".env")) else {
+            throw EnvFileFailure.contentsUnreadable
+        }
+        let lines = contents.split(separator: "\n")
+        // skip comment lines
+        for line in lines where !line.hasPrefix("#") {
+            let pair = line.split(separator: "=")
+            guard let key = pair.first?.trimmingCharacters(in: .whitespacesAndNewlines),
+                  let value = pair.last?.trimmingCharacters(in: .whitespacesAndNewlines),
+                  pair.count == 2,
+                  !key.isEmpty,
+                  !value.isEmpty else {
+                throw EnvFileFailure.malformedKeyValuePair
+            }
+            setenv(key, value, 1)
```

Comment:
Just to be sure @pepicrft @fortmarek , would these be correctly accounted in in the caching logic (I guess yes, as long as they start with `TUIST_`)?

Reply:
That's correct.